File: src/main/java/com/github/bjoernpetersen/spotifyprovider/SpotifyProviderBase.java
package com.github.bjoernpetersen.spotifyprovider;

import com.github.bjoernpetersen.jmusicbot.config.Config;
import com.github.bjoernpetersen.jmusicbot.provider.Provider;
import com.wrapper.spotify.Api;
import javax.annotation.Nonnull;
import org.jetbrains.annotations.NotNull;

public interface SpotifyProviderBase extends Provider {

  @Nonnull
  Api getApi();

  @Nonnull
  Config.ReadOnlyStringEntry getMarket();
}

File: src/main/java/com/github/bjoernpetersen/spotify/provider/SpotifyProviderBase.kt
package com.github.bjoernpetersen.spotify.provider

import com.github.bjoernpetersen.musicbot.api.config.Config
import com.github.bjoernpetersen.musicbot.spi.plugin.Provider
import com.neovisionaries.i18n.CountryCode
import com.wrapper.spotify.SpotifyApi

interface SpotifyProviderBase : Provider {
    //val api: SpotifyApi
    val market: Config.SerializedEntry<CountryCode>
}
